### DISCUSSION TITLE
How to build app file for simulator using Fastlane

### DISCUSSION THREAD

**Discussion by janakirampalepu** [fastlane/fastlane#21996] [General] [answered]
We have requirement to build .app for iOS app for Appium tool to test our application. I would like to know how this can be achieved using Fastlane Script.

Can someone help me with the lane setup required for this requirement.

**Comment by Nelson7896mari** [comment]
Certainly! Setting up a lane in Fastlane to build an iOS .app file for Appium testing can be done with the following steps:

**_Install Fastlane:_**
Make sure you have Fastlane installed on your machine. You can install it via RubyGems if you haven't already:

`sudo gem install fastlane
`

_**Initialize Fastlane:**_
Navigate to your project directory and run the following command to initialize Fastlane:

`fastlane init
`

_**Create a New Lane:**_
Open your Fastfile located in the fastlane directory and define a new lane for building the .app file. You can name the lane as per your preference:

`lane :build_app_for_appium do
  gym(
    export_method: "app",
    include_bitcode: false, # Include if needed
    clean: true,            # Clean the project before building
    workspace: "YourProject.xcworkspace", # Path to your workspace file
    scheme: "YourScheme"    # Name of your scheme
  )
end
`

_**Customize Lane Options:**_
Modify the lane options according to your project setup. Ensure that you specify the correct export method, workspace, and scheme.

_**Run the Lane:**_
Run the newly created lane using the following command:

`fastlane build_app_for_appium
`
_**Verify Output:**_
Once the lane execution is complete, you should find the .app file generated in the output directory specified in your Fastfile.

By following these [HP® Store](https://www.hpinstantink-hp.com) steps, you'll have a Fastlane lane set up to build an iOS .app file suitable for Appium testing. Adjustments can be made to the lane configuration based on your project's specific requirements and settings.

**Reply by beretis** [reply]
@Nelson7896mari this is clearly just an ai response with bs export_method. Fastlane docs states that possible values for export_method are: 
`Valid values are: app-store, validation, ad-hoc, package, enterprise, development, developer-id and mac-application`

**Comment by px-amaac** [comment, marked as answer]
I think this should work. Not AI answer.

```
desc "Create .app file for testing"
  lane :build_app_file do
    gym(
      workspace: "./ios/App.xcworkspace", 
      scheme: "AppScheme",
      configuration: "Release",
      skip_package_ipa: true,
      skip_archive: true,
      build_path: "./output/filePath",
      output_directory: "./output/filePath",
      destination: "generic/platform=iOS Simulator"
    )
    FileUtils.cp_r("../output/filePath/Build/Products/Release-iphonesimulator/AppName.app", "../output")
  end
  ```
  
  - You want a release app. 
  - No export method since its not going out. 
  - Skip the archive and pull the .app file from the build folder.  
  
  Happy Coding.

**Comment by Randall-Developer** [comment]
In case someone else runs into the same issue as I did on a newer version of Fastlane:
Using `skip_archive: true` can result in a ‘No output path received from gym’ error.
Setting `skip_package_ipa: true` alone should be sufficient.
